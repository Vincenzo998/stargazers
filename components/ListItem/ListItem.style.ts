import styled from "styled-components/native";

export const Card = styled.View`
  background-color: #f5f5f5;
  height: 80px;

  flex-direction: row;
  align-items: center;

  padding: 2%;
  margin-horizontal: 10%;
  margin-bottom: 5%;

  border: 0.1px #656565;
  shadow-color: #656565;
  shadow-opacity: 0.2;
  shadow-radius: 5%;
  border-radius: 5%;
`;

export const Avatar = styled.Image`
  border-radius: 50%;
  margin-right: 2%;
`;
