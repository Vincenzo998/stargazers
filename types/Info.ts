// defines
import { REPO, USER } from "../defines";

export type Info = { [USER]: string; [REPO]: string };
