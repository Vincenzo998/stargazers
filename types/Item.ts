export type Item = { id: string; avatar_url: string; login: string };
