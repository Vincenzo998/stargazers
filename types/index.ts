export * from "./ListItemProps";
export * from "./Item";
export * from "./Info";
