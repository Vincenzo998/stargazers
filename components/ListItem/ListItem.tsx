import React, { Text } from "react-native";

// types
import { ListItemProps } from "../../types";

// styles
import { Card, Avatar } from "./ListItem.style";

const ListItem = ({ item: { login, avatar_url, id }, index }: ListItemProps) => {
  return (
    <Card key={id} style={{ ...(index === 0 && { marginTop: "2%" }), shadowOffset: { width: -2, height: 4 } }}>
      <Avatar
        source={{
          height: 50,
          width: 50,
          uri: avatar_url,
        }}
      />
      <Text>{login}</Text>
    </Card>
  );
};

export default ListItem;
