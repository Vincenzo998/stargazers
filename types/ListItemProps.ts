import { Item } from "./Item";

export type ListItemProps = { item: Item; index: number };
