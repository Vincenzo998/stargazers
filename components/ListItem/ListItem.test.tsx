import React from "react";
import renderer from "react-test-renderer";

// screens
import ListItem from ".";

// types
import { Item } from "../../types";

test("ListItem should render without crashing", () => {
  const item: Item = { login: "user", avatar_url: "url", id: "00000-000000-00000" };
  const ListItemComponent = renderer.create(<ListItem item={item} index={0} />).toJSON();

  expect(ListItemComponent).toMatchSnapshot();
});
