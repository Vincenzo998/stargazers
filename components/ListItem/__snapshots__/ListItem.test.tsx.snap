// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render without crashing 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderColor": "#656565",
        "borderRadius": "5%",
        "borderStyle": "solid",
        "borderWidth": 0.1,
        "flexDirection": "row",
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "10%",
        "paddingBottom": "2%",
        "paddingLeft": "2%",
        "paddingRight": "2%",
        "paddingTop": "2%",
        "shadowColor": "#656565",
        "shadowOpacity": 0.2,
        "shadowRadius": "5%",
      },
      {
        "marginTop": "2%",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
      },
    ]
  }
>
  <Image
    source={
      {
        "height": 50,
        "uri": "url",
        "width": 50,
      }
    }
    style={
      [
        {
          "borderRadius": "50%",
          "marginRight": "2%",
        },
      ]
    }
  />
  <Text>
    user
  </Text>
</View>
`;
